assets/app/scripts/directives/oscFileInput.js

```diff
@@ -8,15 +8,83 @@ angular.module('openshiftConsole')
         model: "=",
         required: "=",
         disabled: "=ngDisabled",
-        helpText: "@?"
+        helpText: "@?",
+        dropArea: "@?"
       },
       templateUrl: 'views/directives/osc-file-input.html',
       link: function(scope, element){
         scope.helpID = _.uniqueId('help-');
         scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
         scope.uploadError = false;
+
+        var dropArea = scope.dropArea || element,
+        highlightDropArea = false,
+        showDropArea = false;
+
+        scope.$watch('disabled', function() {
+          if (scope.disabled) {
+            // Remove listeners for dropArea drag & drop actions
+            $(dropArea).off('dragover');
+            $(dropArea).off('dragleave');
+            $(dropArea).off('drop');
+          } else {
+            // Add listeners for dropArea drag & drop actions
+            $(dropArea).on('dragover', function() {
+              $(dropArea).addClass('highlight-drag-and-drop-area');
+              highlightDropArea = true;
+            });
+            $(dropArea).on('dragleave', function() {
+              highlightDropArea = false;
+              _.delay(function(){
+                  if( !highlightDropArea ){ $(dropArea).removeClass('highlight-drag-and-drop-area'); }
+              }, 200 );
```

Comment:
both _.debounce and _.throttle didnt work so I've tried delay which worked.

We set the var to true on `dragover` of the document. On `dragleave` set the flag to false. Then set a brief timeout to check if the flag is still false. The problem here was that if we have nested element in the `dropArea`, each time user drags over the nested element, both `dragleave` and `dragover` are triggered in that order, which will cause a flickering of the `dropArea` border. This way, each dragleave event gives the DOM enough time for a new dragover event to reset the flag. The real, final dragleave that we care about will see that the flag is still false.